### ISSUE TITLE
Support for react 18.2

### ISSUE THREAD

**Issue by alex-kowalczyk** [exah/nano-slots#6]
The package peerDeps specify react 16/17 only. 
This fails on recent npm unless `--legacy-peer-deps` is used.

**Comment by exah** [comment]
Hey @alex-kowalczyk, thanks for reminding me to update the peer deps. 

I've prepared the PR (#7), but I would like to test it before publishing the [stable update](https://github.com/exah/nano-slots/releases/tag/v1.4.0-0). 
Will you be able to try it in a React 18 project?

```
npm install nano-slots@next
```

**Comment by alex-kowalczyk** [comment]
I think this does not cover current react 18.2:
```
npm install --save nano-slots@next
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: test.app@0.1.0
npm ERR! Found: react@18.2.0
npm ERR! node_modules/react
npm ERR!   react@"^18.2.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^16.0.0 || ^17.0.0 | ^18.0.0" from nano-slots@1.4.0-0
npm ERR! node_modules/nano-slots
npm ERR!   nano-slots@"1.4.0-0" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /home/alex/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/alex/.npm/_logs/2022-10-29T13_32_23_037Z-debug-0.log
```

**Comment by exah** [comment]
Thank you! I've made a typo, sorry, please try a new version (install `next` again).

```
npm install --save nano-slots@next
```

**Comment by alex-kowalczyk** [comment]
It installs flawlessly now.

**Comment by exah** [comment]
Released [v1.4.0](https://github.com/exah/nano-slots/releases/tag/v1.4.0)
